File: CS/Customization/WizardCustomizationService.cs
using System.ComponentModel.Design;
using DevExpress.AIIntegration.Reporting;
using DevExpress.AIIntegration.Reporting.Wizard;
using DevExpress.AIIntegration.Reporting.Wizard.Presenters;
using DevExpress.AIIntegration.Reporting.Wizard.Views;
using DevExpress.AIIntegration.WinForms;
using DevExpress.AIIntegration.WinForms.Reporting;
using DevExpress.AIIntegration.WinForms.Reporting.Wizard;
using DevExpress.AIIntegration.WinForms.Reporting.Wizard.Views;
using DevExpress.Data.Utils;
using DevExpress.DataAccess.UI.Wizard;
using DevExpress.DataAccess.Wizard.Model;
using DevExpress.XtraReports.Design;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.Wizards;

namespace AIWizardCustomizationExample.Customization {
    internal class WizardCustomizationService : IWizardCustomizationService {
        public void CustomizeReportWizard(IWizardCustomization<XtraReportModel> tool) {
            tool.StartPage = typeof(ChooseReportCreationModePage<XtraReportModel>);
            tool.RegisterPage<ChooseReportCreationModePage<XtraReportModel>, ChooseReportCreationModePage<XtraReportModel>>();
            tool.RegisterPageView<IChooseReportCreationModePageView, ChooseReportCreationModePageView>();
            tool.RegisterPage<AIChooseDataSourceOptionPage<XtraReportModel>, AIChooseDataSourceOptionPage<XtraReportModel>>();
            tool.RegisterPageView<IAIChooseDataSourceOptionPageView, AIChooseDataSourceOptionPageView>();
            tool.RegisterPage<AINoDataEnterReportPromptPage<XtraReportModel>, AINoDataEnterReportPromptPage<XtraReportModel>>();
            tool.RegisterPageView<IAINoDataEnterReportPromptPageView, AINoDataEnterReportPromptPageView>();
            tool.RegisterPage<AIDataBoundEnterReportPromptPage<XtraReportModel>, AIDataBoundEnterReportPromptPage<XtraReportModel>>();
            tool.RegisterPageView<IAIDataBoundEnterReportPromptPageView, AIDataBoundEnterReportPromptPageView>();
            tool.Model.SetAIParameters(new AIParameters());
            tool.Model.SetPredefinedAIReportPrompts(AIReportPromptCollection.GetDefaultReportPrompts());
        }

        public void CustomizeDataSourceWizard(IWizardCustomization<XtraReportModel> tool) { }

        public bool TryCreateDataSource(IDataSourceModel model, out object dataSource, out string dataMember) {
            dataSource = null;
            dataMember = null;
            return false;
        }

        public bool TryCreateReport(IDesignerHost designerHost, XtraReportModel model, object dataSource, string dataMember) {
            if(model.GetIsAIReportType()) {
                DoWithOverlay(designerHost, () => {
                    var builder = new AIReportBuilder(designerHost, dataSource, dataMember);
                    builder.Build((XtraReport)designerHost.RootComponent, model);
                });
                return true;
            }
            return false;
        }

        void DoWithOverlay(IDesignerHost designerHost, Action action) {
            Control control = designerHost.GetService<ReportTabControlBase>();
            using(var waitForm = new AIOverlayForm()) {
                waitForm.ShowLoading(control);
                try {
                    action();
                    waitForm.Close();
                } catch(Exception ex) {
                    waitForm.ShowError(control, ex.Message, false);
                }
            }
        }
    }
}

File: VB/Customization/WizardCustomizationService.vb
Imports System.ComponentModel.Design
Imports DevExpress.AIIntegration.Reporting
Imports DevExpress.AIIntegration.Reporting.Wizard
Imports DevExpress.AIIntegration.Reporting.Wizard.Presenters
Imports DevExpress.AIIntegration.Reporting.Wizard.Views
Imports DevExpress.AIIntegration.WinForms
Imports DevExpress.AIIntegration.WinForms.Reporting
Imports DevExpress.AIIntegration.WinForms.Reporting.Wizard
Imports DevExpress.AIIntegration.WinForms.Reporting.Wizard.Views
Imports DevExpress.Data.Utils
Imports DevExpress.DataAccess.UI.Wizard
Imports DevExpress.DataAccess.Wizard.Model
Imports DevExpress.XtraReports.Design
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.Wizards

Namespace AIWizardCustomizationExample.Customization

	Friend Class WizardCustomizationService
		Implements IWizardCustomizationService
		Public Sub CustomizeReportWizard(ByVal tool As IWizardCustomization(Of XtraReportModel)) Implements IWizardCustomizationService.CustomizeReportWizard
			tool.StartPage = GetType(ChooseReportCreationModePage(Of XtraReportModel))
			tool.RegisterPage(Of ChooseReportCreationModePage(Of XtraReportModel), ChooseReportCreationModePage(Of XtraReportModel))()
			tool.RegisterPageView(Of IChooseReportCreationModePageView, ChooseReportCreationModePageView)()
			tool.RegisterPage(Of AIChooseDataSourceOptionPage(Of XtraReportModel), AIChooseDataSourceOptionPage(Of XtraReportModel))()
			tool.RegisterPageView(Of IAIChooseDataSourceOptionPageView, AIChooseDataSourceOptionPageView)()
			tool.RegisterPage(Of AINoDataEnterReportPromptPage(Of XtraReportModel), AINoDataEnterReportPromptPage(Of XtraReportModel))()
			tool.RegisterPageView(Of IAINoDataEnterReportPromptPageView, AINoDataEnterReportPromptPageView)()
			tool.RegisterPage(Of AIDataBoundEnterReportPromptPage(Of XtraReportModel), AIDataBoundEnterReportPromptPage(Of XtraReportModel))()
			tool.RegisterPageView(Of IAIDataBoundEnterReportPromptPageView, AIDataBoundEnterReportPromptPageView)()
			tool.Model.SetAIParameters(New AIParameters())
			tool.Model.SetPredefinedAIReportPrompts(AIReportPromptCollection.GetDefaultReportPrompts())
		End Sub

		Public Sub CustomizeDataSourceWizard(ByVal tool As IWizardCustomization(Of XtraReportModel)) Implements IWizardCustomizationService.CustomizeDataSourceWizard
		End Sub

		Public Function TryCreateDataSource(ByVal model As IDataSourceModel, <System.Runtime.InteropServices.Out()> ByRef dataSource As Object, <System.Runtime.InteropServices.Out()> ByRef dataMember As String) As Boolean Implements IWizardCustomizationService.TryCreateDataSource
			dataSource = Nothing
			dataMember = Nothing
			Return False
		End Function

		Public Function TryCreateReport(ByVal designerHost As IDesignerHost, ByVal model As XtraReportModel, ByVal dataSource As Object, ByVal dataMember As String) As Boolean Implements IWizardCustomizationService.TryCreateReport
			If model.GetIsAIReportType() Then
				DoWithOverlay(designerHost, Sub()
												Dim builder = New AIReportBuilder(designerHost, dataSource, dataMember)
												builder.Build(CType(designerHost.RootComponent, XtraReport), model)
											End Sub)
				Return True
			End If
			Return False
		End Function
		Private Sub DoWithOverlay(ByVal designerHost As IDesignerHost, ByVal action As Action)
			Dim control As System.Windows.Forms.Control = designerHost.GetService(Of ReportTabControlBase)()

			Using waitForm = New AIOverlayForm()
				waitForm.ShowLoading(control)

				Try
					action()
					waitForm.Close()
				Catch ex As Exception
					waitForm.ShowError(control, ex.Message, False)
				End Try
			End Using
		End Sub
	End Class
End Namespace
